src/python/zquantum/core/measurement.py

```diff
@@ -412,48 +428,69 @@ def convert_bitstring_to_int(bitstring: Sequence[int]) -> int:
     return int("".join(str(bit) for bit in bitstring[::-1]), 2)
 
 
-def check_parity(
-    bitstring: Union[str, Sequence[int]], marked_qubits: Iterable[int]
-) -> bool:
-    """Determine if the marked qubits have even parity for the given bitstring.
+def check_parity_of_vector(
+    bitstrings_list: List[Union[str, Sequence[int]]], marked_qubits: Iterable[int]
+) -> np.ndarray:
+    """Determine if the marked qubits have even parity for each bitstring in the given list.
 
     Args:
-        bitstring: The bitstring, either as a tuple or little endian string.
+        bitstring: A list of bitstrings that are either tuples or little endian strings.
         marked_qubits: The qubits whose parity is to be determined.
 
     Returns:
-        True if an even number of the marked qubits are in the 1 state, False
-            otherwise.
+        A 1d array with size equal to size of given bitstrings_list. Each entry is 1 if
+        an even number of the marked qubits of the corresponding bitstring are in the 1
+        state and 0 if otherwise.
     """
-    result = True
-    for qubit_index in marked_qubits:
-        if bitstring[qubit_index] == "1" or bitstring[qubit_index] == 1:
-            result = not result
-    return result
+    if not marked_qubits:
+        return np.ones(len(bitstrings_list))
+
+    # Convert bitstrings_list to np array
+    if type(bitstrings_list[0]) is str:
```

Comment:
[request] As mentioned elsewhere, this should be moved outside of the body of this function.